examples/extras_provider.py

```diff
@@ -52,3 +52,13 @@ def get_dependencies(self, candidate):
             req = self.get_base_requirement(candidate)
             deps.append(req)
         return deps
+
+    def narrow_requirement_selection(
```

Comment:
If I understand correctly, implementing this function is optional since it just returns all identifiers by default, right? It’d be good to make this clear in docs, and remove all these stub implementations in examples.